lib/static_html.rb

```diff
@@ -17,12 +17,33 @@
 
 class StaticHtml < Exporter
 
+  TEMPLATE_DIR = File.join(Machinery::ROOT, "html").freeze
+  PARTIAL_DIR = File.join(Machinery::ROOT, "html", "partials").freeze
+
+  include HamlHelpers
+
   def initialize(description)
     @description = description
-    Haml::Engine.new(File.read(File.join(Machinery::ROOT, "html/static_index.html.haml"))).render(Object.new, description: @description)
   end
 
-  def render_scope(partial)
-    Haml::Engine.new(File.read("_#{partial}.html.haml")).render
+  def haml(source, locals:)
```

Comment:
That might interact with Sinatra, because in `lib/server.rb` `haml` is a Sinatra method.